### ISSUE TITLE
IRB hangs when running exec 'ls'

### ISSUE THREAD

**Issue by bharatagarwal** [spacecraft-repl/SpaceCraft#47]
When I try to run `exec 'ls'` in my browser, the repl hangs.

![spacecraft](https://user-images.githubusercontent.com/15229351/50384467-f25ff680-06ea-11e9-92e4-9636780b62ee.gif)

**Comment by bharatagarwal** [comment]
I've tested this out in Firefox Developer Edition and Chrome, both for Mac.

**Comment by YingCGooi** [comment]
Hi @bharatagarwal, based on Ruby docs, the `exec` method *replaces the current process by running the given external command*. It means that the current IRB process will be entirely replaced by the process run by `exec`, and since `ls` immediately exits, there's no running process attached to the REPL terminal. Note that we did not have the feature where the REPL process is restarted. To get around this, just change to another language and change it back to Ruby.

![irb exits](https://i.imgur.com/GPOlPTM.gif)

> The IRB process exits after calling `exec`

**Comment by YingCGooi** [comment]
On a side note: we encourage users to submit PRs if you see a need to add a feature or fix a bug!

**Comment by bharatagarwal** [comment]
Thanks @YingCGooi  for sharing what was going on under the hood, it was great to understand what was causing the issue.

I thought for quite a while over whether this was a bug or a feature. I've done heuristic evaluation for a program as a UX designer previously, and that made me trigger happy in sending issues!.

I think this was very much an edge case, I was messing around with the product at the end of the day. Eventually, I came to the conclusion that I agree with your classification of this as a 'feature request', as it's not something that would stop me from achieving what I needed to do with a REPL.

As for your note about PRs, I agree with the sentiment of it. I'm still some way away from being technically able to supply it myself -- and considering that nobody else is going to feasibly ask for this feature, I'm closing the issue.
